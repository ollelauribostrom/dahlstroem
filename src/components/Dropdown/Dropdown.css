.dropdown {
  position: absolute;
  left: 50%;
  margin-left: -100px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding-top: 15px;
  z-index: 9999;
}

.dropdown__arrow {
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 10px 10px 10px;
  position: absolute;
  top: 5px;
  left: 50%;
  margin-left: -5px;
}

.dropdown__body {
  padding: 15px 20px;
}

.dropdown__item {
  width: 150px;
  overflow: hidden;
  text-decoration: none;
  padding: 5px 0;
  display: block;
}

.dropdown__item:hover {
  text-decoration: underline;
}
