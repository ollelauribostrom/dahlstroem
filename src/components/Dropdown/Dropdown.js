import {Link} from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';

const DropDown = ({categories, onBlur}) => (
  <div className="dropdown" onBlur={onBlur}>
    <div className="dropdown__arrow" />
    <div className="dropdown__body">
      {categories.map (category => (
        <Link to={category.href} className="dropdown__item">
          {category.title}
        </Link>
      ))}
    </div>
  </div>
);

DropDown.PropTypes = {};

export default DropDown;
