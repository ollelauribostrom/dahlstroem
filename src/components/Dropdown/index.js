import Dropdown from './Dropdown';
import './Dropdown.css';

export default Dropdown;
