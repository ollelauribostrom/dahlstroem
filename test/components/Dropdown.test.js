import React from 'react';
import { shallow } from 'enzyme';

import Dropdown from '../../src/components/Dropdown';

describe('Tests for Dropdown', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Dropdown
        categories={[{ href: '/a', title: 'a' }, { href: '/b', title: 'b' }]}
        color={{ hex: '#000' }}
        background={{ hex: '#fff' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
