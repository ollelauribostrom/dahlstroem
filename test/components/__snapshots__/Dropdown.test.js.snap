// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Dropdown renders correctly 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown__arrow"
    style={
      Object {
        "borderColor": "transparent transparent #fff transparent",
      }
    }
  />
  <div
    className="dropdown__body"
    style={
      Object {
        "background": "#fff",
      }
    }
  >
    <mockConstructor
      className="dropdown__item"
      key="/a"
      style={
        Object {
          "color": "#000",
        }
      }
      to="/a"
    >
      a
    </mockConstructor>
    <mockConstructor
      className="dropdown__item"
      key="/b"
      style={
        Object {
          "color": "#000",
        }
      }
      to="/b"
    >
      b
    </mockConstructor>
  </div>
</div>
`;
